torch/_inductor/codegen/cpp.py

```diff
@@ -2092,6 +2092,10 @@ def store_reduction(name, index, value):
 
             @staticmethod
             def constant(val, dtype):
+                if dtype in (torch.float16, torch.bfloat16):
```

Comment:
we are not checking float16 here: https://github.com/pytorch/pytorch/pull/105440/files#diff-5ab7b0235e2076a5fc6629ba0b109208940f5b94f5c13babc3e0f87cf4fcec82R2120. Can float16 work well in the test you added?

Reply:
yes， it can works for my test.